ros/angel_system_nodes/angel_system_nodes/asr.py

```diff
@@ -88,37 +128,31 @@ def listener_callback(self, msg):
                 self.audio_stream_duration = 0.0
 
                 # Start the ASR server request thread.
-                log.info("Spawning ASR request thread.")
+                self.log.info("Spawning ASR request thread.")
                 self.t = threading.Thread(target=self.asr_server_request_thread,
-                    args=(audio_data, msg.channels, msg.sample_rate, self._tmp_wav_filename))
+                    args=(audio_data, msg.channels, msg.sample_rate))
                 self.t.start()
 
-
-    def asr_server_request_thread(self, audio_data, num_channels, 
-        sample_rate, wav_output_filename):
-        log = self.get_logger()
+    def asr_server_request_thread(self, audio_data, num_channels, sample_rate):
         with self.asr_server_lock:
 
-            # TODO(@derekahmed): We should use a temporary audio file instead. Otherwise,
-            # we should at least delete the saved audio file.
-            wf = wave.open(wav_output_filename, 'wb')
+            temp_file = tempfile.NamedTemporaryFile(suffix='.wav', delete=False)
```

Comment:
@derkmed Was this set to `delete=False` for a reason? It seems like this should be `True` instead so that we do not accumulate temporary audio files.

Reply:
Resolved.